app/services/current_team_and_user_service.rb

```diff
@@ -20,6 +20,6 @@ def user
   private
 
   def resolver_for_case(kase)
-    "CurrentTeamAndUser::#{kase.type_abbreviation}::#{kase.workflow.camelize}".constantize.new(kase)
+    "CurrentTeamAndUser::#{kase.type_abbreviation.delete("_")}::#{kase.workflow.camelize}".constantize.new(kase)
```

Comment:
I can't help feeling that perhaps we'd be better off having a method on each case type which defines the current team and user class to use.  Then we wouldn't need lots of empty CurrentTeamAndUser classes.